core/drivers/stpmic1.c

```diff
@@ -587,22 +587,21 @@ bool stpmic1_regulator_is_valid(const char *name)
 	return get_regulator_data(name);
 }
 
-TEE_Result stpmic1_regulator_levels_mv(const char *name,
-				       const uint16_t **levels,
-				       size_t *levels_count)
+int stpmic1_regulator_levels_mv(const char *name,
```

Comment:
If the value is not used this should return `void`. Generally speaking error codes are preferably `TEE_Result` whenever possible.

Reply:
The function implementation returns a value but indeed  the caller does not check it. Would say maybe `pmic_describe_levels`  in` scmi_server.c` should add the check.

Reply:
yes, I would prefer to lease this function as-is and test the return value at caller place.